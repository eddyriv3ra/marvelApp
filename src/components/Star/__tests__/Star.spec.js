import React from "react";
import { shallow } from "enzyme";
import StarComponent from "../Star";

const initialCustomStyle = {
  stroke: "black",
  strokeWidth: 25,
  fill: "gray",
  right: "2rem",
  top: "15%",
};

const onClick = jest.fn();
let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <StarComponent style={initialCustomStyle} onClick={onClick} />
  );
});

describe("StarComponent", () => {
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should handle onClick", () => {
    const event = {
      stopPropagation: jest.fn(),
    };
    wrapper.find("svg").simulate("click", event);
    expect(event.stopPropagation).toHaveBeenCalled();
  });
  it("Should update props styles", () => {
    const newProps = {
      stroke: "black",
      strokeWidth: 25,
      fill: "white",
      right: "2rem",
      top: "15%",
    };
    wrapper.setProps({ style: newProps });
    expect(wrapper.find("svg").props().styleProps).toEqual(newProps);
  });
});
