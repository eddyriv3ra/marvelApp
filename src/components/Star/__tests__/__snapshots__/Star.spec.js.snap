// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarComponent Should render correctly 1`] = `ShallowWrapper {}`;
